src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/ProxyConnectReplyStream.java

```diff
@@ -236,6 +251,54 @@ private void handleEdgeArchSync(
         }
     }
 
+    private void sendCacheableRequest()
+    {
+        CacheableRequest request = (CacheableRequest) streamCorrelation;
+
+        MessageConsumer connect = request.connect();
+        long connectStreamId = request.supplyStreamId().getAsLong();
+        long connectRef = request.connectRef();
+        long connectCorrelationId = request.supplyCorrelationId().getAsLong();
+
+        streamFactory.correlations.put(connectCorrelationId, request);
+        ListFW<HttpHeaderFW> requestHeaders = request.getRequestHeaders(streamFactory.requestHeadersRO);
+        final String etag = request.etag();
+        streamFactory.writer.doHttpBegin(connect, connectStreamId, connectRef, connectCorrelationId,
+                builder ->
+                {
+                    requestHeaders.forEach(
+                            h ->  builder.item(item -> item.name(h.name()).value(h.value())));
+                    if (request instanceof CacheRefreshRequest)
+                    {
+                        builder.item(item -> item.name(HttpHeaders.IF_NONE_MATCH).value(etag));
+                    }
+                });
+        streamFactory.writer.doHttpEnd(connect, connectStreamId);
+        streamFactory.executedRetries.getAsLong();
+    }
+
+    private void retryCacheableRequest(
+            ListFW<HttpHeaderFW> responseHeaders)
+    {
+        long retryAfter = HttpHeadersUtil.retryAfter(responseHeaders);
+
+        CacheableRequest request = (CacheableRequest) streamCorrelation;
+        int retryAttempt = request.incRetryAttemptAndGet();
+        if (retryAttempt > 31)
+        {
+            retryAttempt = 31;
+        }
+        // Exponential backoff based on number of attempts
+        // max(retryAfter, random_between(0, min(cap, base * 2 ^ attempt)))
+        int waitTime = Math.abs(((int) Math.pow(2, retryAttempt) * streamFactory.retryMin));
+        waitTime = streamFactory.random.nextInt(Math.min(streamFactory.retryMax, waitTime));
+        retryAfter = Math.max(retryAfter, waitTime);
+
+        long retryAt = Instant.now().plusMillis(retryAfter).toEpochMilli();
```

Comment:
Perhaps we can move this logic at lines 286-297 inside `CacheableRequest` since the retry attempt number does not otherwise need to be externalized.